1. Listázzuk a <span style="color: #800000;">`kolcsonzes`</span> tábla következő oszlopait: ID, ki\_datum, vissza\_datum! 

  

a.  Az ID oszlop neve legyen Azonosító

b.  A ki\_datum oszlop neve legyen Kölcsönzés\_dátuma

c.  A vissza\_datum oszlop neve legyen Visszahozatal\_dátuma

In [2]:
SELECT ID AS 'Azonosító', 
        ki_datum AS 'A kölcsönzés dátuma',
        vissza_datum 'Visszahozatal dátuma'
        FROM kolcsonzesek;

(5 rows affected)

Total execution time: 00:00:00.055

Azonosító A kölcsönzés dátuma Visszahozatal dátuma 10 2003-12-11 NULL 11 2003-12-11 NULL 12 2003-12-11 NULL 13 2006-10-01 2006-10-10 14 2006-10-02 2006-10-09

2. Melyek azok a kölcsönzések, amelyek 2005.01.01 és 2007.01.01 között történtek?

In [3]:
SELECT ki_datum
FROM kolcsonzesek
--WHERE ki_datum >= '2005.01.01' AND ki_datum <= '2007.01.01'
WHERE ki_datum BETWEEN '2005.01.01' AND '2007.01.01'

(2 rows affected)

Total execution time: 00:00:00.010

ki_datum 2006-10-01 2006-10-02

3. Melyek azok a tagok, akiknek kedvenc stílusa a játék, és nevük a-ra végződik?

  

a.  Csak a tagok neve jelenjen meg, az oszlop neve Név legyen

In [10]:
SELECT nev AS 'Név'
FROM tagok
WHERE kedvencstilus = 'játék'
AND nev LIKE '%a';

(2 rows affected)

Total execution time: 00:00:00.020

Név Bodó Kinga Garay Magdolna

4. Listázzuk egy oszlopban a tagok nevét, címét és kedvenc stílusát összefűzve! 

  

a.  A nevet, címet és a stílust kötőjellel válasszuk el

b.  A kötőjelek előtt és után legyen 1-1 szóköz

c.  Az oszlop neve legyen TAG

d.  Kinga keresztnevű tagok ne legyenek benne a listában

5. Mennyi lenne a dvd-k nettó ára, ha 20%-kal csökkenne? 

  

a.  Csak a dvd-k címét és a csökkentett árat jelenítsük meg

b.  A csökkentett árat nevezzük el diszkont\_ar-nak

In [12]:
SELECT cim, nettoar*0.8 AS 'diszkont_ar'
FROM dvd

(318 rows affected)

Total execution time: 00:00:00.101

cim diszkont_ar NULL 3087,2000000000003 A II. világháború története (1999) 3395,2000000000003 3D Dinosaur - adventure 4032,8 3D virtuális origami (2000) 4284 Az 1848-1849-iki szabadságharc története (2002) 4560 Ablak zsiráf (1999) 2832 Ady Endre összes művei (1999) 3392,8 Afrika 4254,400000000001 Age of Empires (2000) 3585,6000000000004 Airport Inc. 3448,8 American Rodeo : BullRider 2000 (2001) 4472,8 Angol : 8-10 éveseknek (2002) 3882,4 Angol és német nyelvtani gyakorlatok (2002) 3924,8 Angol szótanító CD-ROM 3559,2000000000003 Angol szótanító CD-ROM 3555,2000000000003 Angol szótár (1996) 2807,2000000000003 Angol-magyar, Magyar-angol hangosszótár 4120,8 Angol-magyar nagyszótár (Halász Előd) 4510,400000000001 Az áramvonalas fakanál 4564,8 Arany János: A bajusz (2002) 2712,8 Az Árpád-ház története 4017,6000000000004 Art Attack (1998) 3131,2000000000003 Art puzzle (1998) 4628,8 Az ásványok világa 4499,2 Ausztrália 4260,8 Auto 2000 - Kerékvilág 3570,4 Ázsia (2000) 2498,4 Biblia (1995) 3552 Biológia '97 4434,400000000001 Biológia (2000) 3055,2000000000003 Biológia (2001) 3033,6000000000004 Biológia : 12-14 éveseknek (2003) 2785,6000000000004 Blinky Bill : A fantasztikus léggömbkaland 3260,8 Broken Sword II. (cop. 1997) 3601,6000000000004 Budapest kalauz 3243,2000000000003 Budapest Museum 4233,6 Buggy 4664 Byte Magyarország (2002) 2621,6000000000004 Caesar III. 3308,8 Capitalism II. (2001) 4849,6 Capitalism Plus 4024 Castle Explorer 2965,6000000000004 CD Panoráma 1-4. (1998) 4337,6 Computer iskola (1998) 4208 Csizmás kandúr (1998) 4745,6 Csodálatos univerzum (1997) 3269,6000000000004 Dél-Amerika (1999) 3199,2000000000003 Dementia 1-5. (1997) 4735,2 Deutsch: Anfanger (kezdő) + Telepítő 4506,400000000001 Deutsch: Fortgeschrittene (haladó) + Installation 4627,2 Deutsch: Mittelstufe (középhaladó)+ Telepítő 3501,6000000000004 Diablo 4775,2 Diablo II. (2000) 3920 Dicső napok és hétköznapok 1848 - Digitális forrásgyűjtemény 1. 4488,8 Dreams to Reality 1-2. (1997) 3759,2000000000003 Duke Nukem 3D 4379,2 Dungeon Keeper 2. (1999) 2695,2000000000003 E-Games : Collector's Edition (2001) 4466,400000000001 Éléstár (1998) 2606,4 Élet a polgárosodó Magyarországon (2001) 4457,6 Életképek 1848 (1998) 3368,8 Az ember tragédiája 3436 Encarta World Atlas '99 1-2. (1998) 4267,2 Enemy Engaged RAH-66 4173,6 English 1-2. + Telepítő 4172 English: Intermediate (haladó) 2901,6000000000004 English by Stealth lásd: Lopva angolul 3582,4 Erdély (1996) 3999,2000000000003 Espanol: Intermedio + Telepítő 3131,2000000000003 Espanol: Principiante + Telepítő 4705,6 Észak-Amerika 3195,2000000000003 Európa 4216 The F.A. Premier League Stars 2001 3096,8 Felvételi példatár : Angol 3100,8 Felvételi példatár : Biológia, földrajz 2572 Felvételi példatár : Kémia, fizika 4680,8 Felvételi példatár : Magyar nyelv és irodalom 4764 Felvételi példatár : Matematika 4007,2000000000003 Felvételi példatár : Német 2481,6000000000004 FIFA, 2000 (1999) 4047,2000000000003 Fighting Force (1998) 3840 Final Liberation (1997) 3705,6000000000004 Fitness mesterfokon / Béres Alexandra (2001) 3685,6000000000004 Fizika : 12-14 éveseknek (2002) 2987,2000000000003 Fizika felkészítő az érettségi és felvételi vizsgára 3848 Fizika világa (2001) 4610,400000000001 A Föld élővilága 4720 Földrajz: 10-14 éveseknek (2002) 3144 Földrajz: érettségi előkészítő (2002) 2538,4 Francais 1-2. + Telepítő 3230,4 Francais: Débutan (kezdő) + Telepítő 3016 Francais: Moyen + Telepítő 4495,2 Francia-magyar szótár 2852,8 Funschool in Dreamland (1995) 3812,8 Geolexikon 4.2 : földtani adatbázis (2002) 3681,6000000000004 Gépírás 2000 3148,8 Gondolattár: aforizmák, szállóigék 4276,8 GTA 2 4004 Hadiflották I. (2002) 4099,2 Hadtudományi lexikon (1998) 2812 Hajdúböszörmény 3024,8 Halak, kétéltűek, hüllők 3322,4 Half-Life (1996) 4324 Half-Life Generation: Game of the Year Edition (2002) 2964 Half-Life Generation: Counter Strike (2002) 4595,2 Half-Life Generation: Blue Shift (2002) 4582,400000000001 Harry Potter és a bölcsek köve (2001

6. Melyek azok a kölcsönzések, ahol még nem hozták vissza a dvd-t?

In [15]:
SELECT *
FROM dvd
;

(318 rows affected)

Total execution time: 00:00:00.030

ID cim stilus szam nettoar lang raktár készlet 1 NULL történelem 12342 3859 1 10 2 A II. világháború története (1999) történelem 7224 4244 1 10 3 3D Dinosaur - adventure történelem 576 5041 2 10 4 3D virtuális origami (2000) játék 14235-236 5355 1 10 5 Az 1848-1849-iki szabadságharc története (2002) történelem 13962-963 5700 1 8 6 Ablak zsiráf (1999) ismeretterjesztő 6405 3540 1 10 7 Ady Endre összes művei (1999) irodalom 6334 4241 1 10 8 Afrika földrajz 4248 5318 1 10 9 Age of Empires (2000) játék 14242-244 4482 2 10 10 Airport Inc. játék (repülős) 11026 4311 2 10 11 American Rodeo : BullRider 2000 (2001) játék (rodeó) 10067 5591 2 10 12 Angol : 8-10 éveseknek (2002) nyelvtanító 13293 4853 2 10 13 Angol és német nyelvtani gyakorlatok (2002) nyelvtanító 13794 4906 6 10 14 Angol szótanító CD-ROM nyelvtanító 1722 4449 2 10 15 Angol szótanító CD-ROM nyelvtanító 4202 4444 2 10 16 Angol szótár (1996) szótár 1993 3509 2 10 17 Angol-magyar, Magyar-angol hangosszótár szótár 4724 5151 2 10 18 Angol-magyar nagyszótár (Halász Előd) szótár 7284 5638 2 10 19 Az áramvonalas fakanál gasztronómia 8500 5706 1 10 20 Arany János: A bajusz (2002) irodalom 13424 3391 1 10 21 Az Árpád-ház története történelem 10124 5022 1 10 22 Art Attack (1998) játék 9074 3914 2 10 23 Art puzzle (1998) ismeretterjesztő 2951 5786 2 10 24 Az ásványok világa ásványtan 8473 5624 1 10 25 Ausztrália földrajz 9554 5326 1 10 26 Auto 2000 - Kerékvilág ismeretterjesztő 7873 4463 1 10 27 Ázsia (2000) földrajz 8974 3123 1 10 28 Biblia (1995) ismeretterjesztő 805 4440 1 10 29 Biológia '97 biológia 1161 5543 1 10 30 Biológia (2000) biológia 8982 3819 1 10 31 Biológia (2001) biológia 10318 3792 1 10 32 Biológia : 12-14 éveseknek (2003) biológia 14665 3482 1 10 33 Blinky Bill : A fantasztikus léggömbkaland játék 10195 4076 1 10 34 Broken Sword II. (cop. 1997) játék 1735-1736 4502 2 10 35 Budapest kalauz földrajz 7759 4054 1 10 36 Budapest Museum művészet 7777 5292 1 10 37 Buggy játék 12313 5830 2 10 38 Byte Magyarország (2002) számítástechnika 13993 3277 1 10 39 Caesar III. játék 12322 4136 2 10 40 Capitalism II. (2001) játék 14252 6062 2 10 41 Capitalism Plus játék 1739 5030 2 10 42 Castle Explorer játék (történelmi) 560 3707 2 10 43 CD Panoráma 1-4. (1998) ismeretterjesztő 4294-4374-75-76 5422 1 10 44 Computer iskola (1998) számítástechnika 3742 5260 1 10 45 Csizmás kandúr (1998) játék 5285 5932 1 10 46 Csodálatos univerzum (1997) csillagászat 2128 4087 1 10 47 Dél-Amerika (1999) földrajz 7049 3999 1 10 48 Dementia 1-5. (1997) játék 2675-2679 5919 2 10 49 Deutsch: Anfanger (kezdő) + Telepítő nyelvtanító 8121-8122 5633 3 10 50 Deutsch: Fortgeschrittene (haladó) + Installation nyelvtanító 6976-6977 5784 3 10 51 Deutsch: Mittelstufe (középhaladó)+ Telepítő nyelvtanító 8123-8124 4377 3 10 52 Diablo játék 12321 5969 NULL 10 53 Diablo II. (2000) játék 10039-40-41-42 4900 2 10 54 Dicső napok és hétköznapok 1848 - Digitális forrásgyűjtemény 1. történelem 7793 5611 1 10 55 Dreams to Reality 1-2. (1997) játék 3958-3959 4699 2 10 56 Duke Nukem 3D játék 100 5474 2 10 57 Dungeon Keeper 2. (1999) játék 6831 3369 2 10 58 E-Games : Collector's Edition (2001) játék 9988 5583 2 10 59 Éléstár (1998) gasztronómia 3014 3258 1 10 60 Élet a polgárosodó Magyarországon (2001) történelem 13357 5572 1 10 61 Életképek 1848 (1998) történelem 3874 4211 1 10 62 Az ember tragédiája irodalom 7107 4295 1 10 63 Encarta World Atlas '99 1-2. (1998) ismeretterjesztő 4786-4787 5334 2 10 64 Enemy Engaged RAH-66 játék 11010 5217 2 10 65 English 1-2. + Telepítő nyelvtanító 8118-8119-8120 5215 2 10 67 English: Intermediate (haladó) nyelvtanító 8116-8117 3627 2 10 68 English by Stealth lásd: Lopva angolul NULL NULL 4478 NULL 10 69 Erdély (1996) földrajz 2768 4999 1 10 70 Espanol: Intermedio + Telepítő nyelvtanító 8132-8133 3914 6 10 71 Espanol: Principiante + Telepítő nyelvtanító 8130-8131 5882 6 10 72 Észak-Amerika földrajz 6470 3994 1 10 73 Európa földrajz 7825 5270 1 10 74 The F.A. Premier League Stars 2001 játék (lab

7. Listázzuk azon diákok nevét és születési idejét, akik 1984-ben vagy 1985-ben születtek!  

  

a.  A születési dátum csak az évet, hónapot és a napot tartalmazza! 

b.  A születési dátum oszlopát nevezzük el 'Születési idő'-nek!

In [ ]:
SELECT Nev,

FROM diak

8. Készítsünk lekérdezést, amely a tanulók nevét és az ebből képzett felhasználói nevét tartalmazza. 

  

a.  A felhasználói név álljon a név első kettő, illetve utolsó kettő betűjének összefűzéséből!

b.  A listát szűrjük azon tanulókra, akik teljes neve - szóközzel együtt - legalább 10 betűs!

In [21]:
SELECT Nev, LEFT(Nev,2)+RIGHT(NEV,2)AS 'felhasznalo_nev'
FROM diak
WHERE LEN(Nev)>=10
;

(8 rows affected)

Total execution time: 00:00:00.032

Nev felhasznalo_nev Kovács Imre Kore Horváth Emil Hoil Kapos Petra Kara Csóka Anna Csna Nyúl Tamás Nyás Ordasi Emma Orma Koppány Olga Koga Kozma Patrícia Koia

9. Készítsünk listát a diákok adatairól, ahol a diákok neve úgy jelenik meg, hogy először a keresztnév, majd a vezetéknév, közöttük szóközzel!  

a.  Ezen felül a születési idő három külön oszlopban jelenjen meg (év, hónap, nap). 

b.  Az oszlopokat nevezzük el értelemszerűen!

In [23]:
SELECT *
FROM diak
;

(10 rows affected)

Total execution time: 00:00:00.032

DiakAz Nev Szulido 1 Kos Péter 1987-11-05 00:00:00.000 2 Port Imre 1990-05-05 00:00:00.000 3 Kovács Imre 1984-03-02 00:00:00.000 4 Horváth Emil 1988-05-02 00:00:00.000 5 Kapos Petra 1985-12-23 00:00:00.000 6 Csóka Anna 1981-11-30 00:00:00.000 7 Nyúl Tamás 1988-02-16 00:00:00.000 8 Ordasi Emma 1989-01-03 00:00:00.000 9 Koppány Olga 1984-02-28 00:00:00.000 10 Kozma Patrícia 1983-06-01 00:00:00.000

10. Készítsünk listát a munkák azonosítójáról és megnevezéséről! Egy új oszlopban azt is jelenítsük meg, hogy melyik munka hetente mennyit fizet! 

a.  Az oszlop neve legyen Heti bér, az értéket kerekítsük 1000 forintra! 

b.  A listát szűrjük azon rekordokra, ahol a kerekített heti bér 10000 Ft felett van!

In [ ]:
SELECT munkaId, allas, ROUND(oradij*oraszam*5,-3) AS 'Heti_ber'
FROM munka
WHERE 

11. A diákok számára differenciált béremelést terveznek: a középiskolások esetében 33%, egyéb esetben 17% mértékben. Készítsünk listát, amely tartalmazza a munkák azonosítóját, az állás nevét, az eredeti óradíjat és a tervezett emelt óradíjat.

  

a.  Az oszlopoknak adjuk nevet értelemszerűen!

b.  Az emelt óradíj összegét kerekítsük egészre! (Az esetlegesen megjelenő 0 tizedesjegyekkel ne foglalkozzunk)

12. A diákok számára próbaidőt írnak elő, amely a munka kezdetétől számítva 3 hónapig tart.    Jelenítsük meg a munka tábla adatait egy új oszloppal kiegészítve, amelyik a próbaidő    végének dátumát mutatja!  

  

a.  Az oszlop neve Próbaidő vége legyen! 

b.  A lista ne tartalmazza azokat a munkákat, ahol a diákok azonosítója nincs megadva!

13. Készítsünk listát a munkák azonosítóiról és a megnevezésükről! A megnevezés allas-óraszám formában jelenjen meg, pl: eladó-4. 

  

a.  Jelenítsük meg a dátumot is, de csak a évet és a hónapot, pl: 2003-07! 

b.  Ennek az oszlopnak 'Kezdés hónapja' legyen a neve! A listát szűrjük a nyári hónapokra!

14. Készítsünk listát a diákok adatairól, amely a Nev oszlop helyén a diák monogramját jeleníti meg (keresztnév, illetve vezetéknév első betűje összefűzve). 

a.  A listát szűrjük 1983.01.01 és 1987.06.01 között született diákokra!

15. A diákok a munkakezdés évének utolsó napján bónuszt kapnak, amennyiben az adott évben legalább 6 hónapot dolgoznak. 

  

a.  Jelenítsük meg, hogy az egyes munkák adatait, egy új oszlopban az éves bónusz napját is 'Bónusz dátum' néven! 

b.  A lista csak a bónuszra jogosultak adatait tartalmazza!